rcudnn/cudnn/src/cudnn.rs

```diff
@@ -188,7 +188,7 @@ impl Cudnn {
         hidden_size: i32,
         num_layers: i32,
         seq_length: i32,
-        dropout_desc: &DropoutDescriptor,
+        dropout_memory_ptr: *mut cudnnDropoutStruct,
```

Comment:
this is the solution regarding the memory corruption, correct?

Reply:
while this is totally mergable, we should consider to use a Rc just for this one case, or document the necessity and reasons for it right next to it

Reply:
This change in particular is more about improving visibility on how the Dropout is used.

In the function below, it takes the pointer off the DropoutDescriptor with .id_c(), and then drops the last reference to the DropoutDescriptor, causing a drop of the DropoutDescriptor and removing the CUDA pointer to it. 

In that function there are **lots** of calls to .id_c(), and so it's not obvious that it's only being used for the pointer or that the reference is dropping.

Reply:
I'll take a deeper look later today, I am not sure I understand this sufficiently right now

Reply:
This looks all good to me, I added `b1a0a3eef2ececa303599548284b2edc9c826a53` to a clone of this branch `feat-rnn` which is minimal changeset on top

Reply:
Awesome - was there a memory check on top of it @drahnr, or shall I check that before merging it in?